src/components/FeedbackPrompt.js

```diff
@@ -0,0 +1,143 @@
+  },
+  bugReport: {
+    inputType: Form.TextArea,
+    text: 'If there was a bug or error, describe the bug and what you were trying to do when the bug happened.'
+  },
+  comments: {
+    inputType: Form.TextArea,
+    text: 'Do you have any other comments?'
+  }
+};
+
+/*
+* Modal that shows the feedback form.
+*/
+class FeedbackPrompt extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      formData: {},
+      submissionFailed: false,
+      submitting: false
+    };
+  }
+
+  handleInputChange = (event) => {
+    // Source: https://reactjs.org/docs/forms.html#controlled-components
+    const target = event.target;
+    const value = target.type === 'checkbox' ? target.checked : target.value;
+    const name = target.name;
+
+    const currFormData = this.state.formData;
+    this.setState({
+      formData: Object.assign(currFormData, { [name]: value })
+    });
+  }
+
+  getInputs() {
+    return Object.entries(questions).map(([key, {inputType: InputType, text}]) => {
+        return (
+          <InputType
+            key={key}
+            name={key}
+            label={text}
+            value={this.state.formData[key] || ''}
+            onChange={this.handleInputChange} />
+        );
+      });
+  }
+
+  // returns promise that succeeds if submission is successful, else rejects.
+  sendDataToSpreadsheet(data) {
+    const fetchOptions = {
+      method: 'POST',
+      body: JSON.stringify(data)
+    };
+    return fetch(postUrl, fetchOptions)
+    .then((response) => {
+      if (response.ok) {
+        return true;
+      } else {
+        var error = new Error(response.statusText);
+        error.response = response;
+        throw error;
+      }
+    });
+  }
+
+  cancel = (event) => {
+    // Reset state for next time it's opened
+    this.setState({
+      formData: {},
+      submissionFailed: false,
+      submitting: false
+    });
+    this.props.close();
+  }
+
+  submit = (event) => {
+    this.setState({ submitting: true });
+    const data = Object.assign(this.state.formData, { clientData: this.props.data });
+    this.sendDataToSpreadsheet(data)
+    .then((result) => {
+      this.props.close();
+    })
+    .catch((error) => {
+      this.setState({ submissionFailed: true });
+      console.error(error.message);
+    })
+    .finally(() => {
+      this.setState({ submitting: false });
+    });
+  }
+
+  render () {
+    return (
+      <Modal open={this.props.isOpen}>
+        <Modal.Header>Submit Cliff Effects Feedback</Modal.Header>
+        <Modal.Content>
+          <Form>
+            {this.getInputs()}
+          </Form>
+          <Message hidden={!this.state.submissionFailed} error>
+            Error submitting data, please try again or <a href="mailto:andrew@codeforboston.org">email us</a>.
+          </Message>
+        </Modal.Content>
+        <Modal.Actions>
+          <Button onClick={this.cancel}>Cancel</Button>
```

Comment:
I was thinking actually this is good, since in case the submission is taking a really long time for some reason you wouldn't want to trap the user with the modal open. Closing the modal shouldn't interrupt the submission; the only unexpected behavior I can see happening is submissionFailed gets set after the modal is closed and shows up if it's opened again.

Reply:
If the purpose of that is to allow someone to close the modal when the submission is being sent, then maybe the text for the button should change to 'Done' or 'Close'. Otherwise people will think they're canceling the submission. I suspect they'd end up making the same submission multiple times trying to get it to work.

Reply:
Yeah, I guess. I don't know if we want to be encouraging people to close the modal before they've confirmed that the submission was successful, since if it fails, they can try to resubmit if the modal is still open. If it's taking that long to submit that they decide to click cancel, it was probably failing anyway, so they probably should try to resubmit. 

Maybe we could disable the "cancel" button, but have an x in the corner that doesn't get disabled. And maybe ideally there could be a component visible in the corner or something after closing that reports whether the submission was successful (maybe in the same place as the button to submit feedback, if we decide on a place for that that's visible from all pages?).

Reply:
I like the idea of an 'x' in the corner. Is there a chance the 'cancel' button *could* be made to, in actual fact, cancel a submission while it's being submitted, in case we want them to actually be able to halt the process? We should discuss more about the flow in person, including the successful submission bit. For now in this PR, red 'x' in the corner and 'Cancel' gets disabled when 'Submit' is clicked sounds good to me.

Also, see my very important Slack question.

Reply:
>Is there a chance the 'cancel' button could be made to, in actual fact, cancel a submission while it's being submitted, in case we want them to actually be able to halt the process?

Probably not, since once the request is sent and we're waiting for a response from the Google server, we have no knowledge or control over whether the request will be successful (and if it is, there's no way to undo it (at least without things getting really messy)).

Sounds good re what to do for this PR!